torch/_export/__init__.py

```diff
@@ -106,34 +110,90 @@ def _export(
 
     with torch._dynamo.config.patch(dataclasses.asdict(ExportDynamoConfig())):  # type: ignore[attr-defined]
         try:
-            gm, _ = torch._dynamo.export(
+            gm_torch_level, _ = torch._dynamo.export(
                 f,
                 *args,
-                aten_graph=True,
-                tracing_mode="symbolic",
-                decomposition_table=DECOMP_TABLE,
                 constraints=constraints,
                 assume_static_by_default=True,
-                functionalize=True,
             )
+
+            params = OrderedDict()
+            buffers = OrderedDict()
+
+            for name, param in gm_torch_level.named_parameters(recurse=True, remove_duplicate=False):
+                params[name] = param
+
+            for name, param in gm_torch_level.named_buffers(recurse=True, remove_duplicate=False):
+                buffers[name] = param
+
+            fake_inps = []
+            for node in gm_torch_level.graph.nodes:
+                if node.op == "placeholder" and "val" in node.meta:
+                    fake_val = node.meta["val"]
+                    fake_inps.append(fake_val)
+
+            fake_mode = detect_fake_mode(fake_inps)
+
+            fake_args = pytree.tree_map_only(torch.Tensor, fake_mode.from_tensor, args)
+
+            # Fix the graph output signature to be tuple if scalar
+            # because aot_export expects a tuple as return type
+            return_val = f(*args)
+            out_spec = orig_out_spec = gm_torch_level._out_spec
+            # this means it is scalar return value, so will make it tuple
+            if not isinstance(return_val, (list, tuple, dict)):
+                out_spec = pytree.tree_flatten((return_val,))[1]
+
+            orig_args = gm_torch_level.graph._codegen.pytree_info.orig_args
+
+            gm_torch_level.graph._codegen = _PyTreeCodeGen(
+                _PyTreeInfo(
+                    orig_args,
+                    gm_torch_level._in_spec,
+                    out_spec,
+                )
+            )
+
+            gm_torch_level.recompile()
+            gm, graph_signature = aot_export_module(gm_torch_level, fake_args, decompositions=DECOMP_TABLE, trace_joint=False)
+            mutation_data = MutationData(
```

Comment:
why we want to have a new data structure MutationData instead of using graph_signature directly?

Reply:
There are some fields in GraphSignature such as input_spec and out_spec which is different from what we actually serialize in ExportMetaData. To not confuse users, i decided to filter out things we don't need rn

Reply:
Ok. MutationData looks a bit unintuitive though. We could consider naming it as ExportSignature? 

Or does it make sense to override in_spec and out_spec in GraphSignature? Since the backward_signature is optional and may be used in the future. The benefit is less name to remember hahah.

Reply:
Yeah I agree. Let's sync today to decide on what to do with them.

Reply:
What was the resolution of this meeting?

Reply:
Basically Angela's ExportedProgram diff was the result of this sync.